project-base/tests/ShopBundle/Database/Model/Category/CategoryDomainTest.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Tests\ShopBundle\Database\Model\Category;
+
+use Shopsys\FrameworkBundle\Model\Category\Category;
+use Shopsys\FrameworkBundle\Model\Category\CategoryDataFactory;
+use Shopsys\FrameworkBundle\Model\Category\CategoryFactory;
+use Tests\ShopBundle\Test\DatabaseTestCase;
+
+class CategoryDomainTest extends DatabaseTestCase
+{
+    const FIRST_DOMAIN_ID = 1;
+    const SECOND_DOMAIN_ID = 2;
+    const DEMONSTRATIVE_SEO_TITLE = 'Demonstrative seo title';
+    const DEMONSTRATIVE_SEO_META_DESCRIPTION = 'Demonstrative seo description';
+    const DEMONSTRATIVE_SEO_H1 = 'Demonstrative seo H1';
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Category\CategoryDataFactory
+     */
+    private $categoryDataFactory;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Model\Category\CategoryFactory
+     */
+    private $categoryFactory;
+
+    /**
+     * @var \Doctrine\ORM\EntityManager
+     */
+    private $em;
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->categoryDataFactory = $this->getContainer()->get(CategoryDataFactory::class);
+        $this->categoryFactory = $this->getContainer()->get(CategoryFactory::class);
+        $this->em = $this->getEntityManager();
+    }
+
+    public function testCreateCategoryEnabledOnDomain()
+    {
+        $categoryData = $this->categoryDataFactory->createDefault();
+
+        $categoryData->enabled[self::FIRST_DOMAIN_ID] = true;
+
+        $category = $this->categoryFactory->create($categoryData);
+
+        $refreshedCategory = $this->getRefreshedCategoryFromDatabase($category);
+
+        $this->assertTrue($refreshedCategory->isEnabled(self::FIRST_DOMAIN_ID));
+    }
+
+    public function testCreateCategoryNotEnabledOnDomain()
+    {
+        $categoryData = $this->categoryDataFactory->createDefault();
+
+        $categoryData->enabled[self::FIRST_DOMAIN_ID] = false;
+
+        $category = $this->categoryFactory->create($categoryData);
+
+        $refreshedCategory = $this->getRefreshedCategoryFromDatabase($category);
+
+        $this->assertFalse($refreshedCategory->isEnabled(self::FIRST_DOMAIN_ID));
+    }
+
+    public function testCreateCategoryWithDifferentVisibilityOnDomains()
+    {
+        $categoryData = $this->categoryDataFactory->createDefault();
+
+        $categoryData->enabled = [
+            self::FIRST_DOMAIN_ID => true,
+            self::SECOND_DOMAIN_ID => false,
+        ];
```

Comment:
Here, you are rewriting the whole array, in the other methods you were just rewriting a single item. This inconsistency may have strange consequences when running with different number of domains.

You can achieve what you want without being inconsistent:

```php
$categoryData->enabled[self::FIRST_DOMAIN_ID] = true;
$categoryData->enabled[self::SECOND_DOMAIN_ID] = false;
```

In Payment / Transport tests you even change the number of items - you use single-item arrays.

Not sure if it's a problem that has to be solved...

Reply:
I solved it